script.js

```diff
@@ -1,12 +1,182 @@
+  return `-S${seasonStr}E${episodeStr}`;
 }
 
-window.onload = setup;
+//select one option from select box
+select.addEventListener("change", (event) => {
+  rootElem.textContent = "";
+  showRoot.innerHTML = ""; 
+  lable.innerText = `Displaying 0 / 0 episodes`;
+    searchBar.value = "";
+    const selectedEpisode = event.target.value.slice(-2);
+    const selectedSeason = event.target.value.slice(-5, -3);
+    const selectedMovie = allFetchEpisodes.filter((episode) => {
+      if (
+        episode.season == selectedSeason &&
+        episode.number == selectedEpisode
+      ) {
+        return episode;
+      }
+    });
+    rootElem.innerHTML = makePageForEpisodes(selectedMovie);
+});
+
+
+//level 400:
+//fill the selectShow with all shows
+allShows.sort((a, b) =>
+  a.name.toLowerCase().localeCompare(b.name.toLowerCase())
+);
+allShows.forEach((show) => {
+let option = document.createElement("option");
+option.innerText = `${show.name}`;
+selectShow.appendChild(option);
+})
+
+
+//choose episodes for selected show
+selectShow.addEventListener("change", (event) => {
+  rootElem.textContent = "";
+  lable.innerText = `Displaying 0 / 0 episodes`;
+  let showName = event.target.value;
+
+// Select the show object that matches the selected name
+  let showSelected = allShows.find((show) => show.name === showName);
+  let showId = showSelected.id;
+
+//getting episodes that mach with selected show
+fetchEpisodesForShow(showId);
+})
+
+function fetchEpisodesForShow(showId){ 
+fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+  .then((Response) => Response.json())
+  .then((result) => {
+    makePageForEpisodes(result)
+    select.innerText="";
+    //fill select box to show all episodes ralated to selected show by select show 
+   fillSelectBox(result);
+   allFetchEpisodes = result
+  })
+  }
+
+  //level 500:
+  //make a format in html for shows
+  function makePageForShow(){
+    let output = "";
+    allShows.sort((a, b) =>
+      a.name.toLowerCase().localeCompare(b.name.toLowerCase())
+    );
+    allShows.forEach(show => {
+    
+ output += `
+    <div id="showPage">  
+      
+             <div id="divShowImg" >
+               <img src="${show.image !== null ? show.image.medium : ""}"> 
+             </div>
+             <div id="divShowSummary" >
+               <h1 style="text-align:center;">${show.name}</h1>
+               <p>${show.summary}</p>
+             </div>
+             <div id="divShowUl">
+               <ul>
+                 <li">rated:${show.rating.average}</li>
+                 <li">Genres:${show.genres}</li>
+                 <li>Status:${show.status}</li>
+                 <li>Runtime:${show.runtime}</li>
+               </ul>
+             </div>
+       
+    </div>    
+      
+      `;
+    });
+    showRoot.innerHTML = output;
+  }
+
+  //addEventListener for button to access all shows
+  button.addEventListener("click",(event)=>{
+   event.preventDefault();
+   rootElem.textContent = "";
```

Comment:
When you press back, I would also try to clear the choose episode dropdown and the search bar.